### PR TITLE
amp-bind: bind-impl shouldn't expect form service to exist

### LINKED ISSUE #8812: amp-bind shouldn't expect the ampform service to exist

### PR DESCRIPTION
**Pull request by kmh287**
In order to access amp-form implementations for form elements, bind needs to access the AmpFormService. This PR fixes a bug in bind that expects the service to exist. 

Fixes #8812

### CODE REVIEW

File: extensions/amp-bind/0.1/bind-impl.js [hunk 1]
```diff
@@ -142,8 +145,12 @@ export class Bind {
      * @private {!Promise}
      */
     this.formInitializationPromise_ =
-        ampFormServiceForDoc(this.ampdoc).then(ampFormService => {
-          return ampFormService.whenInitialized();
+        ampFormServiceOrNullForDoc(this.ampdoc).then(ampFormService => {
+          if (ampFormService) {
+            return ampFormService.whenInitialized();
+          } else {
+            return Promise.reject(new Error('Form service not present'));
```
Comment (jridgewell) [inline extensions/amp-bind/0.1/bind-impl.js]: Just throw the error.

Comment (kmh287) [inline extensions/amp-bind/0.1/bind-impl.js]: done

File: extensions/amp-bind/0.1/bind-impl.js [hunk 2]
```diff
@@ -142,8 +145,12 @@ export class Bind {
      * @private {!Promise}
      */
     this.formInitializationPromise_ =
```
Comment (jridgewell) [inline extensions/amp-bind/0.1/bind-impl.js]: Nit: Can we initialize this lazily?

Comment (kmh287) [inline extensions/amp-bind/0.1/bind-impl.js]: sure

File: extensions/amp-bind/0.1/bind-impl.js [hunk 3]
```diff
@@ -838,6 +839,28 @@ export class Bind {
...
+  /*
+   * Form implementations are not filled in until ampdoc is ready.
+   * So we must wait for that to finish before scanning form elements
+   * for dynamic components.
+   * @return {!Promise}
+   * @private
+   */
+  getAmpFormInitializationPromise_() {
+    if (!this.formInitializationPromise_) {
+      const p = ampFormServiceOrNullForDoc(this.ampdoc).then(ampFormService => {
```
Comment (choumx) [inline extensions/amp-bind/0.1/bind-impl.js]: If this returns null, calling `then()` will throw an undefined property error right?

**Pull request closed without merging**